cmd/livepeer/starter/starter.go

```diff
@@ -526,16 +526,21 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 		}
 
 	} else {
-		var keystoreDir string
-		if _, err := os.Stat(*cfg.EthKeystorePath); !os.IsNotExist(err) {
-			keystoreDir, _ = filepath.Split(*cfg.EthKeystorePath)
-		} else {
-			keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
-		}
+		var keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
+		if *cfg.EthKeystorePath != "" {
+			keystoreInfo, err := ParseEthKeystorePath(*cfg.EthKeystorePath)
+			if keystoreInfo.path.(string) != "" {
+				keystoreDir = keystoreInfo.path.(string)
+				if keystoreInfo.address.(string) != "" && err == nil {
+					*cfg.EthAcctAddr = keystoreInfo.address.(string)
```

Comment:
I think this is a tricky part, because from what I understand now, if you have both `EthKeystorePath` and `EthAcctAddr` defined then, livepeer node will use the address from keystore.

The best would be to fail fast if you have `EthKeystorePath` and `EthAcctAddr` defined with different ETH addresses. But this can be difficult to check, then in that case, I think we should maybe not allow specifying `EthAcctAddr` if your `EthKeystorePath` points directly to a file. What do you think?

Reply:
I created standard formatting for the ETH addresses by using the `ethcommon.HexToAddress` function to santize input, which returns a '0x0' eth address when invalid. EthAcctAddr is only updated if it is empty (flag not provided) or `EthAcctAddr` and `keystoreInfo.address` do not match. Added log message for mismatched addresses.